### PR TITLE
Implement Calculations in the file

### PR DESCRIPTION
**Pull request by camilevahviraki**
### **In this pull request we:**

- Implemented the calculation function in `Calculator.js`
- Added the `calculate.js` and `operate` files
- Updated the design

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,38 +1,112 @@
...
+  }
+
+dropValue = (e) => {
+  const X = e.target.value.toString();
+
+  if (X === '-' || X === '+' || X === '÷' || X === 'x' || X === '%') {
+    this.setState({ nextString: !this.state.nextString });
+    this.setState({ operation: e.target.value });
+    this.setState({ outputStr: '0' });
+    console.log('condition', 'A');
```
Comment (TedLivist) [inline src/components/Calculator.js]: - [ ] Kindly remove unnecessary console logs

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,38 +1,112 @@
...
+    } else {
+      this.setState({ total: `${this.state.total}${e.target.value}` });
+      this.setState({ outputStr: this.state.total + X });
+      console.log('condition', 'C');
+    }
+  } else if ((X.match(/^[0-9]+$/) !== null && this.state.nextString === true) || X === '.') {
+    if (this.state.next === null) {
+      this.setState({ next: X });
+      this.setState({ outputStr: X });
+      console.log('condition', 'D');
```
Comment (TedLivist) [inline src/components/Calculator.js]: - [ ] Kindly remove console logs. It kills performance.

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,38 +1,112 @@
...
+    console.log('condition', 'F');
+  } else if (X === '+/-') {
+    if (this.state.next !== null) {
+      this.setState({ next: `${-1 * parseInt(this.state.next, 10)}` });
+      this.setState({ outputStr: `${-1 * parseInt(this.state.outputStr, 10)}` });
+      console.log('condition', 'G');
+    } else {
+      this.setState({ total: `${-1 * parseInt(this.state.total, 10)}` });
+      this.setState({ outputStr: `${-1 * parseInt(this.state.outputStr, 10)}` });
+      console.log('condition', 'H');
```
Comment (TedLivist) [inline src/components/Calculator.js]: - [ ] Kindly remove console logs

File: src/components/Calculator.js [hunk 4]
```diff
@@ -1,38 +1,112 @@
...
+    next: this.state.next,
+    operation: this.state.operation,
+  };
+
+  if (calculate(obj, X).total && X === '=') {
+    this.setState({ outputStr: calculate(obj, X).total });
+    this.setState({ total: null });
+    this.setState({ next: null });
+    this.setState({ nextString: false });
+    console.log('condition', 'I');
```
Comment (TedLivist) [inline src/components/Calculator.js]: - [ ] Kindly remove console logs

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @camilevahviraki,

Good job so far!
✔️ Linters are passing
✔️ Events are implemented
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] **[OPTIONAL]** I would recommend you pay attention to your console and fix this warnings in order to write more React-friendly code

  warnings

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
Hi  @camilevahviraki 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] n/a

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**